pkg/scannerv4/mappers/mappers.go

```diff
@@ -834,6 +847,13 @@ func vulnerabilityName(vuln *claircore.Vulnerability) string {
 		if v, ok := findName(vuln, awsVulnNamePattern); ok {
 			return v
 		}
+	// TODO(ROX-26672): Remove this to show CVE as the vuln name.
+	case strings.EqualFold(vuln.Updater, rhelUpdaterName), strings.EqualFold(vuln.Updater, rhccUpdaterName):
+		if !features.ScannerV4RedHatCVEs.Enabled() {
+			if v, ok := findName(vuln, rhelVulnNamePattern); ok {
+				return v
+			}
+		}
```

Comment:
The name returned here will be used to match NVD enrichments, at line #338. It was OK before when NVD was optional and we assumed RHSA should have scores -- now that we need to return both, we need to take that into account.

Reply:
It does look like the test case `when Red Hat CVEs disabled return RHSA` tests for this scenario to an extent, when the RHSA is used the 'first' `nvdCVE` is picked.  

![image](https://github.com/user-attachments/assets/68e190fa-18a6-4977-b9d0-37c5e42fc919)

Assuming there is a scenario where the 'first' cve is not be correct then agreed should use the actual `CVE` identifier when determining the `nvdVuln`.

Reply:
I understand where you're coming from, you're missing the fact that golang's map does not guarantee order, and that CVE enrichments are matched not only in the vulnerability name, but in the description, and links.

Reply:
I updated things above and will followup higher up in the file ^